rust/datafusion/src/optimizer/utils.rs

```diff
@@ -416,6 +424,7 @@ pub fn rewrite_expression(expr: &Expr, expressions: &Vec<Expr>) -> Result<Expr>
                 Ok(expr)
             }
         }
+        Expr::InList { .. } => Ok(expr.clone()),
```

Comment:
likewise here, I think we might want to include the `list` -- even though at the moment it only contains constants, it is a `Vec<Expr>`

Reply:
Here this is just cloning the while `InList` expression (not the `expr` in `InList`) as the optimiser is not doing anything for this Expression yet.